terminateTestProcesses.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/sh
+
+# Arg S1 : Unix current env.User
+
+# If Windows uses PowerShell
+if [ "$OSTYPE" = "cygwin" ]; then
+  echo "Windows machine, using powershell queries..."
+
+  # Match anything that is most likely a test job or process related
+  match_str="(CommandLine like '%java%jck%' or \
+             CommandLine like '%openjdkbinary%java%' or \
+             CommandLine like '%java%javatest%' or \
+             CommandLine like '%java%-Xfuture%' or \
+             CommandLine like '%X%vfb%' or \
+             CommandLine like '%rmid%' or \
+             CommandLine like '%rmiregistry%' or \
+             CommandLine like '%tnameserv%' or \
+             CommandLine like '%make.exe%')"
+
+  # Ignore Jenkins agent and grep cmd
+  ignore_str="not CommandLine like '%remoting.jar%' and \
+              not CommandLine like '%agent.jar%' and \
+              not CommandLine like '%grep%'"
+
+  count=`powershell -c "(Get-WmiObject Win32_Process -Filter {${ignore_str} and ${match_str}} | measure).count" | tr -d "\\\\r"`
+  
+  if [ $count -gt 0 ]; then
+      echo Windows rogue processes detected, attempting to stop them..
+      powershell -c "Get-WmiObject Win32_Process -Filter {${ignore_str} and ${match_str}}"
+      powershell -c "(Get-WmiObject Win32_Process -Filter {${ignore_str} and ${match_str}}).Terminate()"
+      uname | grep CYGWIN 2>/dev/null || uptime
+      echo Sleeping for 10 seconds...
+      sleep 10
+      uname | grep CYGWIN 2>/dev/null || uptime
+      count=`powershell -c "(Get-WmiObject Win32_Process -Filter {${ignore_str} and ${match_str}} | measure).count" | tr -d "\\\\r"`
+      if [ $count -gt 0 ]; then
+        echo "Cleanup failed, ${count} processes still remain..."
+        exit 127
+      fi 
+      echo "Processes stopped successfully"
+  else
+      echo Woohoo - no rogue processes detected!
+  fi
+else
+  echo "Unix type machine.."
+
+  # Match anything that is most likely a jck test job or process related
+  match_str="java.*jck|openjdkbinary.*java|java.*javatest|java.*-Xfuture|X.*vfb|rmid|rmiregistry|tnameserv|make"
+
+  # Ignore Jenkins agent and grep cmd
+  ignore_str="remoting.jar|agent.jar|grep"
+
+  if [ `uname` = "SunOS" ]; then
```

Comment:
Note: This seems to have changed from the previous version. The new one will likely not work on Solaris 11.
(This code looks familiar ;-) I'm going to have to look separately on adjusting this at the source to be ok on Solaris 11!)

Reply:
ok thanks?

Reply:
@sxa maybe other places that use uname SunOS check need changing? https://github.com/adoptium/aqa-tests/blob/d089897c5c9a7d87ba6def8efd53a4cb786c2e99/buildenv/jenkins/JenkinsfileBase#L425

Reply:
Ah ok! yes, so this and your other scripts will need altering for Sol11

Reply:
updated